devito/tools.py

```diff
@@ -56,9 +63,22 @@ def aligned(a, alignment=16):
     extra = alignment / a.itemsize
     buf = np.empty(a.size + extra, dtype=a.dtype)
     ofs = (-buf.ctypes.data % alignment) / a.itemsize
-    aa = buf[ofs:ofs+a.size].reshape(a.shape)
+    aa = buf[ofs:ofs + a.size].reshape(a.shape)
     np.copyto(aa, a)
 
     assert (aa.ctypes.data % alignment) == 0
 
     return aa
+
+
+def get_cache_line_size():
+    """Gets cache line size
+    :returns: cache line size"""
+    if "Linux" in platform.platform():
+        with open('/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size') as f:
+            return int(f.read())
+    else:  # Running on MAC
+        command = "/usr/sbin/sysctl -a | grep hw.cachelinesize"
```

Comment:
`sysctl` did not seem to find anything related to cache line size on Richter. If you mean MAC i think @ggorman confirmed that i worked on his map, is that true?
We tried to envoke `sysctl` through `ctypes` is python however that turned out to be a massive pain and the other alternative was to call compiled c script  that @pp1336  found which would return all processor related information. However, I think we agreed that it was too much of an over kill for such simple problem.